source/common/common/token_bucket_impl.h

```diff
@@ -29,8 +32,10 @@ class TokenBucketImpl : public TokenBucket {
   const double max_tokens_;
   const double fill_rate_;
   double tokens_;
+  absl::optional<bool> reset_once_;
   MonotonicTime last_fill_;
   TimeSource& time_source_;
+  absl::Mutex mutex_;
```

Comment:
If you're going to introduce a mutex to guard variables or functions, it would be good to use `ABSL_GUARDED_BY(...)` where relevant.